src/sentry/search/events/builder/metrics.py

```diff
@@ -162,13 +163,16 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
                 Organization.objects.get_from_cache(id=self.organization_id),
             )
 
+            spec_version = (
+                get_spec("use_updated_env_logic") if use_updated_env_logic else get_spec()
```

Comment:
This is a little clunky.

Reply:
Definitely, it's the "drawback" of the approach. It's a bit of a backward logic to me to look for the spec version with a specific flag but it's the easiest thing to do.

Another idea, which is a bit weird still, is to build a phantom spec version with the flags we need for querying.